iothub_client/samples/iothub_client_c2d_streaming_proxy_sample/iothub_client_c2d_streaming_proxy_sample.c

```diff
@@ -107,13 +111,22 @@ static void on_bytes_received(void* context, const unsigned char* buffer, size_t
 
     if (g_is_uws_client_ready)
     {
-        if (uws_client_send_frame_async(g_uws_client_handle, 2, buffer, size, true, on_ws_send_frame_complete, NULL) != 0)
+        size_t send_size;
+
+        while (size > 0)
         {
-            (void)printf("Failed sending data to the local service\r\n");
-            g_continueRunning = false;
-        }
+            send_size = size <= MAX_WEBSOCKET_PAYLOAD_SIZE_IN_BYTES ? size : MAX_WEBSOCKET_PAYLOAD_SIZE_IN_BYTES;
 
-        g_outgoingByteCount += size;
+            if (uws_client_send_frame_async(g_uws_client_handle, 2, buffer, send_size, true, on_ws_send_frame_complete, NULL) != 0)
+            {
+                (void)printf("Failed sending data to the local service\r\n");
+                g_continueRunning = false;
```

Comment:
>g_continueRunning = false; [](start = 16, length = 26)

Shouldn't you break out here?  If you get a fail it would continue and this might not be desirable.